Kianoosh76/IBS
# [IBS-66] Task endpoint for instructors

# Description
This PR includes changes related to the following task-related endpoints:
- `all_tasks_staff`
- `add_task_staff`
- `get_task_staff`
- `delete_tasks_staff`
- `change_task_staff`

Changes include modifications to both endpoint implementations, their respective seeder files, and test case files.

## Changes Made
- Updated Task Model and Migration file to add a belong to relation to TaskGroup model
- Added constraint to the Task Model
- Tiny changes made to two queries for functions `weight_validate` and `task_validate` in  `utilities/helpers.js` 

## Test Guidelines
1. Run the following series of commands
```
git pull
git checkout ibs-92-rename-testcase
docker system prune -a && docker volume rm $(docker volume ls -q)
make build && make run
```
2. Ensure that all tests pass before merging this PR. In `backend/` run the following commands
```
npm test all_tasks_staff
npm test add_task_staff
npm test get_task_staff
npm test delete_tasks_staff
npm test change_task_staff
```
